### ISSUE TITLE
Storybook 6: No props found for ts component using `PropsWithChildren`

### ISSUE THREAD

**Issue by zmdavis** [storybookjs/storybook#11013]
**Describe the bug**
When using `PropsWithChildren` to help define the type for a functional component used in a story, the `Props` component returns "No props found for this component".

**To Reproduce**
Given a component such as:

```
import React, { FC, PropsWithChildren } from "react";

type ButtonProps = {
  /**
   * Simple click handler
   */
  onClick?: () => void;
};

/**
 * The world's most _basic_ button
 */
export const Button: FC<ButtonProps> = ({ children, onClick }: PropsWithChildren<ButtonProps>) => (
  <button onClick={onClick} type="button">
    {children}
  </button>
);
```

And a simple stories file:

```
import React from 'react';

import { Button } from './Button';

export default {
  component: Button,
  title: 'Components/Button',
};

export const Basic = () => <Button>My Button</Abbr>;
```

The generated docs file will display "No props found for this component".

If the component is updated to not use `PropsWithChildren`, props are rendered as expected:

```
import React, { FC, ReactNode } from "react";

type ButtonProps = {
  children: ReactNode;
  /**
   * Simple click handler
   */
  onClick?: () => void;
};

/**
 * The world's most _basic_ button
 */
export const Button: FC<ButtonProps> = ({ children, onClick }: ButtonProps) => (
  <button onClick={onClick} type="button">
    {children}
  </button>
);
```

**Expected behavior**
Props should be found regardless of use of `PropsWithChildren`, as was the behavior when using `react-docgen-typescript-loader` in Storybook 5.3.x. 

**System:**
```
  System:
    OS: macOS 10.15.4
    CPU: (16) x64 Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
  Binaries:
    Node: 13.8.0 - /usr/local/bin/node
    Yarn: 1.22.0 - /usr/local/bin/yarn
    npm: 6.13.7 - /usr/local/bin/npm
  Browsers:
    Chrome: 83.0.4103.61
    Firefox: 75.0
    Safari: 13.1
  npmPackages:
    @storybook/addon-actions: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/addon-docs: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/addon-knobs: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/addon-storyshots: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/addons: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/react: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/source-loader: 6.0.0-beta.20 => 6.0.0-beta.20 
    @storybook/theming: 6.0.0-beta.20 => 6.0.0-beta.20
```

**Comment by shilman** [comment]
@zmdavis interesting. the zero-config typescript setup uses `react-docgen-typescript-loader`. Any chance you can run `yarn storybook --debug-webpack` on the working project and the broken one and tease out the key difference?

**Comment by zmdavis** [comment]
Was finally able to come back to this, and it appears to be fixed in `v6.0.0-beta.32`.  Thanks!
